src/Microsoft.AspNet.Mvc.TagHelpers/CacheTagHelper.cs

```diff
@@ -346,11 +351,22 @@ private void AddVaryByRouteKey(StringBuilder builder)
             }
         }
 
-        private static IEnumerable<string> Tokenize(string value)
+        private static IList<string> Tokenize(string value)
         {
-            return value.Split(AttributeSeparator, StringSplitOptions.RemoveEmptyEntries)
-                        .Select(token => token.Trim())
-                        .Where(token => token.Length > 0);
+            var values = value.Split(AttributeSeparator, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
Sadly, yes it does. As for an issue, already had you covered: https://github.com/aspnet/Razor/issues/623